pdns/dnsname.cc

```diff
@@ -41,12 +41,15 @@ std::ostream & operator<<(std::ostream &os, const DNSName& d)
   return os <<d.toLogString();
 }
 
-DNSName::DNSName(const char* p, size_t length)
+DNSName::DNSName(const std::string_view sw)
 {
-  if(p[0]==0 || (p[0]=='.' && p[1]==0)) {
+  const char* p = sw.data();
+  size_t length = sw.length();
```

Comment:
It's not the smartest change, but maybe cleaning the internals is worth an other PR, it open a room for improving.

I don't feel confident enough in cpp to do it quickly.